src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,54 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_USER_AGE = 18;
+    private static final int MINIMUM_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        user = checkUserNotNull(user);
+        user = checkUserLogin(user);
+        user = checkUserAge(user);
+        user = checkUserPassword(user);
+        if (user != null) {
+            storageDao.add(user);
+            return user;
+        }
+        return null;
+    }
+
+    private User checkUserNotNull(User user) {
+        if (user == null) {
+            throw new RuntimeException("user cannot be null");
+        }
+        return user;
+    }
+
+    private User checkUserLogin(User user) {
+        if (user.getLogin() == null) {
+            throw new RuntimeException("user login cannot be null");
+        }
+        if (storageDao.get(user.getLogin()) == null) {
+            return user;
+        }
```

Comment:
```suggestion
        if (storageDao.get(user.getLogin()) != null) {
            throw new RuntimeException("a user with this login already exists");
        }
```
Your task in these private methods is to check if the user object is valid and if it is not, throw an exception. Fix it everywhere